packages/@cdktf/provider-generator/lib/get/generator/loop-detection.ts

```diff
@@ -0,0 +1,83 @@
+import { AttributeModel } from "./models/attribute-model";
+
+type AttributeIdentifier = [string, Set<string>];
+const equalAttributeIdentifiers = (
+  x: AttributeIdentifier,
+  y: AttributeIdentifier
+) =>
+  x[0] === y[0] &&
+  x[1].size === y[1].size &&
+  [...x[1]].every((x) => y[1].has(x));
+
+function typeStructure(model: AttributeModel) {
+  if (model.type.isPrimitive) {
+    return model.type.name;
+  }
+
+  return "<complex>";
```

Comment:
This seems like it has the potential for a false match if attribute names on different levels match, but they have different value types.

Reply:
I completely agree with you, will improve it in a follow up, I want to check this against the prebuilt provider